lib/mxcreply.cpp

```diff
@@ -90,10 +91,17 @@ MxcReply::MxcReply()
 
 qint64 MxcReply::readData(char *data, qint64 maxSize)
 {
-    return d->m_device->read(data, maxSize);
+    const qint64 size = d->m_device->read(data, maxSize);
+    d->m_bytesRead += size;
+    return size;
 }
 
 void MxcReply::abort()
 {
     d->m_reply->abort();
 }
+
+qint64 Quotient::MxcReply::bytesAvailable() const
+{
+    return d->m_reply->bytesAvailable() + d->m_bytesRead;
```

Comment:
Actually, I guess this should work, instead of trying to track the number of read bytes?
```suggestion
    return d->m_device->bytesAvailable() + QNetworkReply::bytesAvailable();
```
Making sure to call the parent's `bytesAvailable()` is what the Qt documentation on `QIODevice::bytesAvailable()` instructs to do, by the way - to take account that internal buffer to which the first `readData()` puts the peeked chunk.

Reply:
Actually that's perfect! It contains the right amount of bytes which it peeked. I'll amend the MR with those changes, along with using the correct device.